### ISSUE TITLE
Worker kill not worked

### ISSUE THREAD

**Issue by Ekstazi** [amphp/parallel#63]
Hi. I wrote simple test code:
```php
		$task = new \application\Test();
		\Amp\Loop::repeat(100, function($reference) use($task){
			\Amp\Loop::disable($reference);
			$pool = \Amp\Parallel\Worker\pool();
			echo "Get worker";
			$worker = $pool->getWorker();
			try {
				echo 'enqueue task';
				$promise = $worker->enqueue($task);
				$canceller = new \Amp\TimeoutCancellationToken(1000);
				$id = $canceller->subscribe(function() use($worker){
					echo "Killing";
					$worker->kill();
				});
				echo "Waiting";
				$result = yield $promise;
				echo "done";
				$canceller->unsubscribe($id);
			} catch(Throwable $exception) {
				echo $exception->getMessage();
			}
			echo 'Result is:' . $result;
			if(!$worker->isRunning()) {
				echo "Worker not running";
			}
			\Amp\Loop::enable($reference);
		});
		\Amp\Loop::run();

```
and Test is:
```php
class Test implements Task
{
	/**
	 * @inheritDoc
	 */
	public function run(Environment $environment)
	{
		sleep(30);
		return 'ok';
	}

}

```

I expect what worker will be killed after 1 second and "enqueue" promise will fail. But instead of that promise succeeds after 30 seconds.

**Comment by Ekstazi** [comment]
Related bug:
https://github.com/amphp/process/issues/29

**Comment by trowski** [comment]
Fix in [1.0.3](
https://github.com/amphp/process/releases/tag/v1.0.3) of `amphp/process`.
